src/test/scala/is/hail/distributedmatrix/RowMatrixSuite.scala

```diff
@@ -1,11 +1,76 @@
 package is.hail.distributedmatrix
   
+import breeze.linalg.DenseMatrix
 import is.hail.SparkSuite
 import is.hail.utils._
 import org.testng.annotations.Test
 
-
 class RowMatrixSuite extends SparkSuite {
+  private def rowArrayToRowMatrix(a: Array[Array[Double]]): RowMatrix = {
+    require(a.length > 0)
+    val nRows = a.length
+    val nCols = a(0).length
+    
+    RowMatrix(hc, sc.parallelize(a.zipWithIndex.map { case (row, i) => (i.toLong, row) }), nCols, nRows)
+  }
+  
+  private def rowArrayToLocalMatrix(a: Array[Array[Double]]): DenseMatrix[Double] = {
+    require(a.length > 0)
+    val nRows = a.length
+    val nCols = a(0).length
+    
+    new DenseMatrix[Double](nRows, nCols, a.flatten, 0, nCols, isTranspose = true)
+  }
+  
+  @Test
+  def localizeRowMatrix() {
+    val fname = tmpDir.createTempFile("test")
+    
+    val rowArrays = Array(
+      Array(1.0, 2.0, 3.0),
+      Array(4.0, 5.0, 6.0))
+
+    val rowMatrix = rowArrayToRowMatrix(rowArrays)
+    val localMatrix = rowArrayToLocalMatrix(rowArrays)
+    
+    BlockMatrix.from(hc.sc, localMatrix).write(fname)
+    
+    assert(rowMatrix.toLocalMatrix() === localMatrix)
+  }
+
+  @Test
+  def readBlockSmall() {
+    val fname = tmpDir.createTempFile("test")
+    
+    val localMatrix = DenseMatrix(
+      Array(1.0, 2.0, 3.0),
+      Array(4.0, 5.0, 6.0))
+    
+    BlockMatrix.from(hc.sc, localMatrix).write(fname, forceRowMajor = true)
+    
+    val rowMatrixFromBlock = RowMatrix.readBlockMatrix(hc, fname, 1)
+    
+    assert(rowMatrixFromBlock.toLocalMatrix() == localMatrix)
+  }
+  
+  @Test def readBlock() {
+    val fname = tmpDir.createTempFile("test")
+    val r = scala.util.Random
+    r.setSeed(0)
+    
+    val lm = DenseMatrix.fill[Double](9, 10)(r.nextDouble())
```

Comment:
I think we should use `Gen.denseMatrix`:

```scala
Prop.forAll(Gen.denseMatrix(9, 10)) { lm => 
  ...
}
```

I wouldn't make the blockSize/partSize random though, I like that we exhaustively check all those combinations.